website/blog/weaver_generate.md

```diff
@@ -0,0 +1,499 @@
+func (router) Put(_ context.Context, key, value string) string { return key }
+```
+
+Next, we embed `weaver.WithRouter[router]` in the `cache` struct:
+
+```go
+type cache struct {
+    weaver.Implements[Cache]
+    weaver.WithRouter[router]
+    ...
+}
+```
+
+With all this in place, `Cache` method calls with the same key will be routed to
+the same replica of the `Cache` component.
+
+You may forget to run `weaver generate` after adding, removing, or changing an
+embedded `weaver.WithRouter[T]`.
+
+```diff
+ type cache struct {
+     weaver.Implements[Cache]
+-    weaver.WithRouter[router]
+ }
+```
+
+To detect this, we introduce a `weaver.RoutedBy[T]` interface and
+`weaver.Unrouted` interface.  If a component struct embeds
+`weaver.WithRouter[T]`, then it implements `weaver.RoutedBy[T]`. All other
+component structs implement `weaver.Unrouted`.
+
+```go
+// Component A is routed.
+type a struct {
+    weaver.Implements[A]
+    weaver.WithRouter[router]
+}
+
+// Component B is not routed.
+type b struct {
+    weaver.Implements[B]
+}
+
+var _ weaver.RoutedBy[router] = (*a)(nil)
+var _ weaver.Unrouted = (*b)(nil)
+```
+
+First, we implement `weaver.RoutedBy[T]` by adding an unexported `routedBy(T)`
+method to `weaver.WithRouter[T]`:
+
+```go
+type WithRouter[T any] struct{}
+func (WithRouter[T]) routedBy(T) {}
+```
+
+Then, we define the `weaver.RoutedBy[T]` interface.
+
+```go
+type RoutedBy[T any] interface {
+    routedBy(T)
+}
+```
+
+Similar to `weaver.InstanceOf[T]`, because `routedBy(T)` is unexported, only
+structs that embed `weaver.WithRouter[T]` will implement the
+`weaver.RoutedBy[T]` interface.
+
+Next, we implement the `weaver.Unrouted` interface. This is a bit tricky because
+we want a component that *doesn't* embed `weaver.WithRouter` to implement
+`weaver.Unrouted`. How do we make a struct implement an interface based on the
+absence of something? We start by defining an `implementsImpl` struct that
+implements the `weaver.RoutedBy[private]` interface for an empty unexported
+`private` type:
+
+```go
+type private struct{}
+type implementsImpl struct{}
+func (implementsImpl) routedBy(private) {}
+```
+
+Next, we embed `implementsImpl` in the `weaver.Implements[T]` struct:
+
+```go
+type Implements[T any] struct {
+    implementsImpl
+}
+```
+
+With this in place, any component implementation struct that embeds
+`weaver.Implements[T]` will inherit the `routedBy(private)` method and therefore
+implement `weaver.RoutedBy[private]`.
+
+
+```go
+type cache struct {
+    weaver.Implements[Cache]
+    ...
+}
+
+// cache inherits the routedBy(private) method from the embedded
+// weaver.Implements[Cache].
+var _ func(private) = cache{}.routedBy
+```
+
+We can then define `weaver.Unrouted` as an alias of `weaver.RoutedBy[private]`:
+
+```go
+type Unrouted interface {
+    routedBy(private)
+}
+```
+
+Now, every component struct that embeds `weaver.Implements` implements the
```

Comment:
Do you need this paragraph, it's obvious 🙄 .

Just kidding, this is great reading for hard-core folks!

Reply:
Haha, yeah this stuff gets pretty hard-core!